src/NodaTime.Demo/ZonedDateTimeDemo.cs

```diff
@@ -39,5 +40,183 @@ public void SkippedLocalDateTime()
             DateTimeZone dublin = DateTimeZoneProviders.Tzdb["Europe/Dublin"];
             Assert.Throws<SkippedTimeException>(() => dublin.AtStrictly(new LocalDateTime(2010, 3, 28, 1, 15, 0)));
         }
+
+        [Test]
+        public void TickOfDay()
+        {
+            // This is a 25-hour day at the end of daylight saving time
+            var dt = new LocalDate(2017, 10, 29);
+            var time = new LocalTime(23, 59, 59);
+            var dublin = DateTimeZoneProviders.Tzdb["Europe/Dublin"];
+
+            var startOfDay = dublin.AtStartOfDay(dt);
+            ZonedDateTime nearEndOfDay = dublin.AtStrictly(dt + time);
+
+            Snippet.For(nearEndOfDay.TickOfDay);
+            Assert.AreEqual(time.TickOfDay, nearEndOfDay.TickOfDay);
+
+            Duration duration = nearEndOfDay - startOfDay;
+            Assert.AreEqual(Duration.FromHours(25) - Duration.FromSeconds(1), duration);
+
+            Assert.AreNotEqual(duration.TotalTicks, nearEndOfDay.TickOfDay);
+        }
+
+        [Test]
+        public void AddDurationAndIsDaylightSavingTime()
+        {
+            // This is the date when DST ended at 2am in the given time zone.
+            var dt = new LocalDateTime(2017, 10, 29, 1, 45, 0);
+
+            DateTimeZone dublin = DateTimeZoneProviders.Tzdb["Europe/Dublin"];
+
+            ZonedDateTime beforeTransition = new ZonedDateTime(dt, dublin, Offset.FromHours(1));
+
+            Assert.AreEqual(true, Snippet.For(beforeTransition.IsDaylightSavingTime()));
+            var result = Snippet.For(ZonedDateTime.Add(beforeTransition, Duration.FromHours(1)));
+            Assert.AreEqual(new LocalDate(2017, 10, 29), result.Date);
```

Comment:
A comment would be useful here: "Adding an hour of elapsed time takes us across the DST transition, so we have the same local time (shown on a clock) but a different offset" or similar.